app.rb

```diff
@@ -25,16 +26,31 @@ def initialize
   end
 
   get '/' do
-    erb :index, locals: { projects: key_list }
+    if authed?
+      erb :index, locals: { projects: key_list }
+    else
+      redirect '/auth/myusa'
+    end
+  end
+
+  get '/auth/myusa/callback' do
+    if env['omniauth.auth']
+      session[:authenticated] = true
```

Comment:
The session is secure, right? i.e. someone couldn't just set this themselves? Also, I'd maybe store something like `user_email` instead, which I could see being useful in the future.